massa-protocol-worker-2/src/tests/ban_nodes_scenarios.rs

```diff
@@ -0,0 +1,568 @@
+// Copyright (c) 2022 MASSA LABS <info@massa.net>
+
+use std::time::Duration;
+
+use massa_consensus_exports::test_exports::MockConsensusControllerMessage;
+use massa_models::{block_id::BlockId, prehash::PreHashSet, slot::Slot};
+use massa_protocol_exports_2::{test_exports::tools, ProtocolConfig};
+use massa_signature::KeyPair;
+use massa_time::MassaTime;
+use peernet::peer_id::PeerId;
+use serial_test::serial;
+
+use crate::{
+    handlers::{
+        block_handler::{BlockInfoReply, BlockMessage},
+        operation_handler::OperationMessage,
+    },
+    messages::Message,
+    wrap_network::ActiveConnectionsTrait,
+};
+
+use super::{context::protocol_test, tools::assert_hash_asked_to_node};
+
+#[test]
+#[serial]
+fn test_protocol_bans_node_sending_block_header_with_invalid_signature() {
+    let mut protocol_config = ProtocolConfig::default();
+    protocol_config.thread_count = 2;
+    protocol_config.initial_peers = "./src/tests/empty_initial_peers.json".to_string().into();
+    protocol_test(
+        &protocol_config,
+        move |mut network_controller,
+              protocol_controller,
+              protocol_manager,
+              mut consensus_event_receiver,
+              pool_event_receiver| {
+            //1. Create 1 node
+            let node_a_keypair = KeyPair::generate();
+            let (node_a_peer_id, _node_a) = network_controller.create_fake_connection(
+                PeerId::from_bytes(node_a_keypair.get_public_key().to_bytes()).unwrap(),
+            );
+
+            //2. Create a block with bad public key.
+            let mut block = tools::create_block(&node_a_keypair);
+            block.content.header.content_creator_pub_key = KeyPair::generate().get_public_key();
+            //end setup
+
+            //3. Send header to protocol.
+            network_controller
+                .send_from_peer(
+                    &node_a_peer_id,
+                    Message::Block(Box::new(BlockMessage::BlockHeader(
+                        block.content.header.clone(),
+                    ))),
+                )
+                .unwrap();
+
+            std::thread::sleep(std::time::Duration::from_millis(1000));
+            //4. Check that node is banned.
```

Comment:
in fact that check only if connection with remote peer is closed

Reply:
yes no problem, just update the comment like `Check that node connection is closed (node should be banned)`

Reply:
ok